frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -113,6 +113,9 @@ public int getNumRunningWorkers(String modelName) {
             if (minWorker == 0) {
                 threads = workers.remove(model.getModelName());
                 if (threads == null) {
+                    if (maxWorker == 0) {
```

Comment:
I am not sure if `maxWorker==0` check is correct. Could you test the following sequence:1
1. Register a model (no initial workers)
2. Scale up the workers (maybe to 2 workers)
3. Scale down the workers to 0.
4. Scale up the workers again (maybe to 2 workers)..
Minworkers and Maxworkers are always the same number. So, when we scale down to 0, according to this change we would be remvoing the server thread. When will the server thread again be created? Its initially created during registration of the model.

Reply:
This was failing even before the fix. If you initialiaze a model with workers > 0, scale down to 0 and scale up again, we get an exception with serverthread being null. So essentially, there is no change in behaviour.
I can create a separate issue to track this and will send out another PR.

Reply:
Created a separate issue to fix this bug https://github.com/awslabs/multi-model-server/issues/916

Reply:
Thanks for raising the issue. Its definitely a bug that needs fixing. So currently scaledown to 0 and scaleup is broken. Only work around would be `DELETE /models` and `POST /models`.